src/components/FiltersBar/FiltersBar.tsx

```diff
@@ -0,0 +1,501 @@
+import { add } from 'date-fns'
+import React, { useCallback, useEffect, useState } from 'react'
+import { useParams } from 'react-router'
+import { CSSTransition } from 'react-transition-group'
+
+import { VideoOrderByInput, VideoWhereInput } from '@/api/queries'
+import { SORT_OPTIONS } from '@/config/sorting'
+import knownLicenses from '@/data/knownLicenses.json'
+import { useMediaMatch } from '@/hooks/useMediaMatch'
+import { Button } from '@/shared/components/Button'
+import { Checkbox } from '@/shared/components/Checkbox'
+import { PopoverDialog } from '@/shared/components/Popover'
+import { RadioButton } from '@/shared/components/RadioButton'
+import { Select } from '@/shared/components/Select'
+import { Text } from '@/shared/components/Text'
+import { SvgGlyphClose } from '@/shared/icons'
+
+import {
+  FilterContentContainer,
+  FiltersContainer,
+  FiltersInnerContainer,
+  MobileFilterContainer,
+  OtherFilterStyledIcon,
+  OtherFilterStyledText,
+} from './FiltersBar.styles'
+
+import { ActionDialog, ActionDialogProps } from '../ActionDialog'
+import { StyledTitleText } from '../MessageDialog/MessageDialog.style'
+
+type VideoLengthOptions = '0-to-4' | '4-to-10' | '10-to-9999'
+
+export const FiltersBar: React.FC<ReturnType<typeof useFiltersBar>> = ({
+  handleApplyFilter,
+  filters: {
+    setiIsFiltersOpen,
+    isFiltersOpen,
+    sortVideosBy,
+    setSortVideosBy,
+    dateUploadedFilter,
+    setdateUploadedFilter,
+    matureContentRatingFilter,
+    setMatureContentRatingFilter,
+    paidPromotionalMaterialFilter,
+    setPaidPromotionalMaterialFilter,
+    videoLegnthFilter,
+    setVideoLegnthFilter,
+    licensesFilter,
+    setLicensesFilter,
+  },
+  canClearFilters: {
+    canClearDateUploadedFilter,
+    canClearVideoLegnthFilter,
+    canClearOtherFilters,
+    canClearAllFilters,
+    clearAllFilters,
+    clearDateUploadedFilter,
+    clearVideoLegnthFilter,
+    clearLicensesFilter,
+    clearOtherFilters,
+  },
+}) => {
+  const smMatch = useMediaMatch('sm')
+  const betweenBaseAndSMMatch = !smMatch
+
+  const handleSorting = (value?: VideoOrderByInput | null) => {
+    if (value) {
+      setSortVideosBy(value)
+    }
+  }
+
+  const dateUploadedInputs = (
+    <FilterContentContainer>
+      <RadioButton
+        onChange={() => {
+          setdateUploadedFilter(1)
+        }}
+        name="date-uploaded"
+        label="Last 24 hours"
+        value={1}
+        selectedValue={dateUploadedFilter}
+      ></RadioButton>
```

Comment:
If the JSX children are empty, let's not use both opening and closing tag. This can be a self-closing tag instead (`<RadioButton />`). This applies to a several tags in this file